crates/core/src/router.rs

```diff
@@ -145,6 +152,32 @@ impl Router {
         }
     }
 
+    fn select_closest_peers<'a>(
+        &self,
+        peers: impl IntoIterator<Item = &'a PeerKeyLocation>,
+        target_location: &Location,
+    ) -> Vec<&'a PeerKeyLocation> {
+        let mut heap = std::collections::BinaryHeap::with_capacity(self.closest_peers_capacity + 1);
+
+        for peer_location in peers {
+            if let Some(location) = peer_location.location.as_ref() {
+                let distance = target_location.distance(location);
+                heap.push((distance, peer_location));
+
+                // Ensure we keep the heap size to specified capacity
+                if heap.len() > self.closest_peers_capacity {
+                    heap.pop();
+                }
+            }
+        }
+
+        // Convert the heap to a sorted vector
+        heap.into_sorted_vec()
```

Comment:
The into_sorted_vec method first converts the heap into a vector and then sorts it. This could be less efficient than directly popping elements off the heap. Instead of converting the heap into a sorted vector and then iterating over it, you could pop elements from the heap until it is empty. This approach directly benefits from the heap's properties and might be more efficient, especially if self.closest_peers_capacity is significantly smaller than the number of peers.

This new vector will have a known size so we should allocate it `with_capacity`.

This function will be called very frequently so efficiency is an important consideration.

Reply:
This would be a good catch in most circumstances, but actually for BinaryHeap I think this is very efficient, because it's just a wrapper around Vec so this is prettry much free to do.

Reply:
Ah, you're right - my bad.